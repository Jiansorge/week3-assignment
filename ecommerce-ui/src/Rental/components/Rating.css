.rating-info {
    display:flex;
    flex-flow: row wrap;
    justify-content: flex-start;
    font-size:0.65rem;
    padding: 2px 0 4px ;
}

.stars{
    color:green;

    font-weight:900;
}