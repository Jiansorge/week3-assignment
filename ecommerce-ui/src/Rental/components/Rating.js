//  - rating stars
//  - rating reviews
//  - is Superhost

import React from 'react';
import './Rating.css'
import PropTypes from 'prop-types';

class Rating extends React.Component {
  static propTypes = {
    rental: PropTypes.object.isRequired,
  }

    render() {

      return (
        <div className = "rating-info">
          <p className="stars">
            {this.props.rental.rating.stars} stars
         </p>
         <p className="reviews">
            &nbsp;
            {this.props.rental.rating.reviews} reviews
          </p>
          <div className={"superhost"}>
            { //Check if host has superHost
                        (this.props.rental.host.isSuperhost )
                            ? <div> 
                                &nbsp; &#8226; &nbsp; Superhost
                              </div> 
                            : ""
                }
          </div>
        </div>
      );
    }

  }
  export default Rating;