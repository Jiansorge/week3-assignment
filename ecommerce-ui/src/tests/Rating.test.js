import React from 'react';
import Rating from '../Rental/components/Rating.js';
import {shallow} from 'enzyme';
import airbnbs from '../airbnbs.json';


//import testData from './testData.json';

// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<Image />, div);
//   ReactDOM.unmountComponentAtNode(div);
// });

it('should match the snapshot', () => {
  const wrapper = shallow(<Rating rental={airbnbs[0]} />);
  expect(wrapper).toMatchSnapshot();
});