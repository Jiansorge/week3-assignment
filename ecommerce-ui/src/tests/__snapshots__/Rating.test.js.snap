// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div
  className="rating-info"
>
  <p
    className="stars"
  >
    3
     stars
  </p>
  <p
    className="reviews"
  >
     
    133
     reviews
  </p>
  <div
    className="superhost"
  >
    <div>
        •   Superhost
    </div>
  </div>
</div>
`;
